dataflow_test.py

```diff
@@ -168,17 +168,26 @@ def test_count():
 
     assert count.future.result() == len(the_source)
 
+# 'push' provides a higher-level interface to using such futures:
+# it optionally accepts a future, a tuple of futures or a mapping
+# of futures. It returns the result of the future, a tuple of
+# their results or a namespace with the results, respectively.
```

Comment:
I'll need to take some time to calmly think about whether we can unite these three tests. The crucial differences seem drowned in the noise of the setup ... but I don't immediately see how this can be mitigated.